devlab/tests/test_resource_migration.py

```diff
@@ -428,3 +429,16 @@ def test_ext_router_ip_changed(self):
                     'GW ip addresses of router "{0}" are same on src and dst:'
                     ' {1}'.format(dst_router['name'],
                                   dst_gateway['fixed_ips'][0]['ip_address']))
+
+    def test_not_valid_vms_did_not_migrate(self):
+        all_vms = self.migration_utils.get_all_vms_from_config()
+        vms = [vm['name'] for vm in all_vms if vm.get('broken')]
+        migrated_vms = []
+        for vm in vms:
+            try:
+                self.dst_cloud.get_vm_id(vm)
+                migrated_vms.append(vm)
+            except generate_load.NotFound:
```

Comment:
`generate_load.NotFound` looks very strange, I assume there should be no dependencies between generation script and tests. Maybe moving exception type to a separate module would help?